redisvl/index.py

```diff
@@ -19,6 +19,12 @@
 from redisvl.utils.utils import check_redis_modules_exist, convert_bytes, make_dict
 
 
+def _default_process_results(results: List["Result"]) -> List[Dict[str, Any]]:
```

Comment:
Couple thoughts

1. I think  a ``processor`` class with a ``__call__`` method would be clearer here. and shorter ha
2. should we take out some of the fields like ``payload`` by default? the ``__dict__`` method here usually returns alot of cruft.